### PR TITLE
[FIXED JENKINS-41478] Add button should be enabled if any stores are enabled

### PR DESCRIPTION
**Pull request by stephenc**
- Also disabled stores should be disabled in the menu

See [JENKINS-41478](https://issues.jenkins-ci.org/browse/JENKINS-41478)

@reviewbybees

### CODE REVIEW

File: src/main/java/com/cloudbees/plugins/credentials/CredentialsSelectHelper.java [hunk 1]
```diff
@@ -180,6 +180,37 @@ public ModelObject resolveContext(Object context) {
...
+        if (includeUser) {
+            User current = User.current();
+            if (current != null && current.hasPermission(CREATE)) {
+                return true;
+            }
+        }
+        if (context == null) {
+            StaplerRequest request = Stapler.getCurrentRequest();
+            if (request != null) {
+                context = request.findAncestorObject(ModelObject.class);
```
Comment (jglick) [inline src/main/java/com/cloudbees/plugins/credentials/CredentialsSelectHelper.java]: And if that is null also, do we gracefully return false, or does `lookupStores` give an NPE?

**Comment by stephenc** [comment]
screen shot 2017-01-26 at 16 21 25

Here is the Add button now enabled even though the current user does not have create permission at the root element, but does in the folder

Note that the folder is selectable while the root store is disabled as it is read-only to the current user

File: src/main/resources/lib/credentials/select.jelly [hunk 2]
```diff
@@ -87,13 +87,24 @@
...
+                  <li class="yuimenuitem" data-url="${request.contextPath}/${storeItem.url}/dialog">
+                    <span class="yuimenuitemlabel" tooltip="${storeItem.description}">
+                      <l:icon class="${storeItem.iconClassName} icon-sm"/>
+                      ${storeItem.displayName}
+                    </span>
+                  </li>
+                </j:when>
+                <j:otherwise>
+                  <li class="yuimenuitem" disabled="disabled">
+                    <span class="yuimenuitemlabel" tooltip="${storeItem.description}" disabled="disabled">
```
Comment (jglick) [inline src/main/resources/lib/credentials/select.jelly]: Would make for a smaller diff and simpler code to introduce a variable for `storeItem.enabled ? null : 'disabled'` and use it on both the `li` and the `span`.

File: src/main/resources/lib/credentials/select/select.js [hunk 3]
```diff
@@ -200,6 +200,17 @@ Behaviour.specify("BUTTON.credentials-add-menu", 'credentials-select', -99, func
             }
             window.credentials.add(item.srcElement.getAttribute('data-url'));
         });
+        // YUI menu will not parse disabled when using DIV-LI only when using SELECT-OPTION
```
Comment (jglick) [inline src/main/resources/lib/credentials/select/select.js]: Oof. Will we ever be able to replace YUI without having to go all the way to Blue Ocean?

**Pull request merged**